NdagiStanley/bucketlist
# Task 0 - Create API (CRUD)

#### What does this PR do?

Have REST API working
#### Description of Task to be completed?

Implement the following structure:

``` json
{
  "id": 1,
  "name": "BucketList1",
  "items": [
    {
      "id": 1,
      "name": "I need to do X",
      "date_created": "2015-08-12 11:57:23",
      "date_modified": "2015-08-12 11:57:23",
      "done": false
    }
  ],
  "date_created": "2015-08-12 11:57:23",
  "date_modified": "2015-08-12 11:57:23",
  "created_by": "1113456"
}
```
#### How should this be manually tested?

Upon cloning and cd'ing into the repo:
(on the terminal)
- `cd peewee-orm`
- RUN `python api.py start`
- RUN `python api.py runserver`

Switch to Postman then RUN the following

`http://localhost:5000/` (index page) (GET)

`http://localhost:5000/bucketlists/` endpoint (GET)

`http://localhost:5000/bucketlists/1/items/` endpoints (GET)

---

To use sqlalchemy orm which will be the ORM in use forthwith:

`cd sqlalchemy_orm` or if you were in peewee_orm directory `cd ..` then `cd sqlalchemy_orm`
- RUN `python api.py start`
- RUN `python api.py runserver`

Switch to Postman then RUN the following

`http://localhost:5000/`(index page) (GET)

`http://localhost:5000/bucketlists/` endpoint (GET, POST)

in POST the body should be in this format:

``` json
{
  "name": "Travel",
  "created_by": "user"
}
```

`http://localhost:5000/bucketlists/1` endpoints (GET, PUT, DELETE)

the body for PUT here is same as the above body :point_up_2:

`http://localhost:5000/bucketlists/1/items` endpoints (POST)

in POST the body should be in this format:

``` json
{
  "title": "Travel"
}
```

`http://localhost:5000/bucketlists/1/items` endpoints (PUT, DELETE)

in PUT the body should be in this format:

``` json
{
  "title": "Travel to Mombasa",
  "done": true
}
```
###### Worth noting: _DELETE_ and _GET_ have no body
